Feature/ae 1234 polttoaineet

muep opened the pull request.

muep committed:
AE-1234 Split out generation of a single test ET

While writing tests for AE-1234, it seemed useful to have convenient
access to generating a single ET for testing.

muep committed:
AE-1234 Test for search by ostetut polttoaineet

muep committed:
AE-1234 Fix search by ostettu polttoaine

The value in the raw Energiatodistus needs to be multiplied by the
energy content factor that is specific to each ostettu polttoaine.

Note: The factors likely should be shared with the ones found in
etp-backend/src/main/clj/solita/etp/service/complete_energiatodistus.clj

muep commented on etp-backend/src/main/clj/solita/etp/service/energiatodistus_search_fields.clj:
```diff
@@ -124,9 +131,27 @@
+      (ostettu-polttoaine-entry [:toteutunut-ostoenergiankulutus
+                                 :ostetut-polttoaineet
+                                 :pilkkeet-havu-sekapuu]
+                                1300
```
Nämä taikanumerot pitäisi jemmata johonkin

hkorpi replied:
ne voisi laittaa esim. solita.etp.service.polttoaine/muunnoskertoimet {:kevyt-polttooljy 10 ....}
ja näiden pohjalta luoda noi rivit tänne ja complete-energiatodistukseen

hkorpi replied:
nämä vakiothan on käytössä molemmissa

hkorpi replied:
tuohon funktioon on aika turha enää tuota skeemaa laittaa kun sitähän on käytetty vain noiden kenttien löytämiseen automaattisesti

hkorpi replied:
sen sijaan noi saisi näppärästi generoitua tuosta: solita.etp.service.polttoaine/muunnoskertoimet objektista

muep replied:
Otetaan sieltä myös tuo tuplen jälkimmäiseen lokeroon menevä schemasiivu, mutta ihan totta että voi sen valinnan tehdä esim tuon funktion ulkopuolellakin.

muep commented on etp-backend/src/test/clj/solita/etp/service/energiatodistus_search_test.clj:
```diff
@@ -322,6 +322,52 @@
+    (t/is (contains? (->> (service/public-search
+                           ts/*db*
+                           kayttaja-test-data/paakayttaja
+                           {:where [[["=" "energiatodistus.toteutunut-ostoenergiankulutus.ostetut-polttoaineet.kevyt-polttooljy-neliovuosikulutus"
```
Vähän arveluttaa tämä `=` -haku numeeriselle suureelle, mutta sattuu osumaan niin että tarkalla arvolla haku onnistuu.

muep committed:
AE-1234 Share common energy coefficients

Both the computed search fields and the corresponding values in
complete-energiatodistus need the same coefficients, which are now
located in solita.etp.service.polttoaine/muunnosokertoimet.

hkorpi commented on etp-backend/src/main/clj/solita/etp/service/complete_energiatodistus.clj:
```diff
@@ -441,10 +442,9 @@
-          (assoc-in [:toteutunut-ostoenergiankulutus :ostetut-polttoaineet :puupelletit-kerroin] 4.7)
+          (update-in [:toteutunut-ostoenergiankulutus :ostetut-polttoaineet]
+                     (partial merge (map/map-keys #(keyword (str (name %) "-kerroin"))
+                                                  polttoaine/muunnoskertoimet)))
```
:+1:
